client/store/actions.js

```diff
@@ -14,13 +14,18 @@ export default {
 
             commit('setUser', { data });
         })
-        .catch(function (error) {
-            console.log(error);
+        .catch((error) => {
+            
+            const { response: {
+                data
+            } } = error;
+
+            commit('setError', { data })
         });
     },
 
     userRegister({commit}, payload) {
-        const { email, password, username, file } = payload
+        const { email, password, password_repeat, username, file } = payload
```

Comment:
So you add it here and removed in the `Register.vue`. Why?  In addition you never used it in this action.